src/main/java/org/candlepin/subscriptions/security/OptInChecker.java

```diff
@@ -46,6 +46,11 @@ public OptInChecker(OrgConfigRepository orgConfigRepository) {
 
   public boolean checkAccess(Authentication authentication) {
     Object principal = authentication.getPrincipal();
+    if (RhAssociatePrincipal.class.isAssignableFrom(principal.getClass())
+        || PskClientPrincipal.class.isAssignableFrom(principal.getClass())
+        || (principal instanceof String principalString && principalString.equals("self"))) {
```

Comment:
I see this condition is to allow access when: 

```
curl 'http://localhost:8000/api/rhsm-subscriptions/v1/instances/billing_account_ids?org_id=6261679' -H x-rh-swatch-psk:placeholder
```

However, don't see why we should support this (don't see the requirement in the JIRA ticket). 
Note that adding this condition here, will affect also to the other API we have...